rust/src/lib.rs

```diff
@@ -106,430 +139,387 @@ pub extern "C" fn zcashlc_clear_last_error() {
-            Err(e) => Err(format_err!("Error while initializing accounts: {}", e)),
-        }
-    });
-    unwrap_exc_or(res, false)
+/// Frees a FFIBinaryKey value
+///
+/// # Safety
+///
+/// - `ptr` must be non-null and must point to a struct having the layout of [`FFIBinaryKey`].
+///   See the safety documentation of [`FFIBinaryKey`].
+#[no_mangle]
+pub unsafe extern "C" fn zcashlc_free_binary_key(ptr: *mut FFIBinaryKey) {
+    if !ptr.is_null() {
+        let key: Box<FFIBinaryKey> = Box::from_raw(ptr);
+        let key_slice: &mut [u8] = slice::from_raw_parts_mut(key.encoding, key.encoding_len);
+        drop(Box::from_raw(key_slice));
+    }
 }
 
-/// Derives Extended Spending Keys from the given seed into 'accounts' number of accounts.
-/// Returns the ExtendedSpendingKeys for the accounts. The caller should store these
-/// securely for use while spending.
+/// Adds the next available account-level spend authority, given the current set of [ZIP 316]
+/// account identifiers known, to the wallet database.
+///
+/// Returns the newly created [ZIP 316] account identifier, along with the binary encoding of the
+/// [`UnifiedSpendingKey`] for the newly created account.  The caller should manage the memory of
+/// (and store) the returned spending keys in a secure fashion.
+///
+/// If `seed` was imported from a backup and this method is being used to restore a
+/// previous wallet state, you should use this method to add all of the desired
+/// accounts before scanning the chain from the seed's birthday height.
+///
+/// By convention, wallets should only allow a new account to be generated after funds
+/// have been received by the currently-available account (in order to enable
+/// automated account recovery).
+///
+/// # Safety
+///
+/// - `db_data` must be non-null and valid for reads for `db_data_len` bytes, and it must have an
+///   alignment of `1`. Its contents must be a string representing a valid system path in the
+///   operating system's preferred representation.
+/// - The memory referenced by `db_data` must not be mutated for the duration of the function call.
+/// - The total size `db_data_len` must be no larger than `isize::MAX`. See the safety
+///   documentation of pointer::offset.
+/// - `seed` must be non-null and valid for reads for `seed_len` bytes, and it must have an
+///   alignment of `1`.
+/// - The memory referenced by `seed` must not be mutated for the duration of the function call.
+/// - The total size `seed_len` must be no larger than `isize::MAX`. See the safety documentation
+///   of pointer::offset.
+/// - Call [`zcashlc_free_binary_key`] to free the memory associated with the returned pointer when
+///   you are finished using it.
 ///
-/// Call `zcashlc_vec_string_free` on the returned pointer when you are finished with it.
+/// [ZIP 316]: https://zips.z.cash/zip-0316
 #[no_mangle]
-pub unsafe extern "C" fn zcashlc_derive_extended_spending_keys(
+pub extern "C" fn zcashlc_create_account(
```

Comment:
```suggestion
pub unsafe extern "C" fn zcashlc_create_account(
```

Reply:
Deferring this to https://github.com/zcash-hackworks/zcash-light-client-ffi/issues/65#issuecomment-1315988668.